### ISSUE TITLE
Algorithm Problem 

### ISSUE THREAD

**Issue by PragueSquare** [EMResearch/EvoMaster#4]
Hello, I've been studying this project for a few days and I think it's great.  However, I have a problem which puzzles me. The WTS Algorithm is mainly introduced in the paper "RESTful API Automated Test Case Generation" , but there is no real implementation of WTS Algorithm in the source code you provided（the implementation in the file WTSAlgorithm.kt is only experiments). And the source code is mainly about the implementation of MIO algorithm.Could you please tell me why?

**Comment by arcuri82** [comment]
Hi, 
 WTSAlgorithm is a full working implementation of WTS. I pushed some clarifications on the documentation of that class

**Comment by PragueSquare** [comment]
Thank you very much for your answer. I read your paper "Seeding strategies in search-based unit test generation" and tried to use "constant seeding" strategy. However, I ran into a problem.

As the following example,
       if (x > 10) {
             ...
       }
How do we collect the constant of "10"?

The paper says that the constants can be obtained by bytecode, but is there a value "10"  in the constant pool of the bytecode? I can get the variable "x" through ASM now, but I can't get the "value" 10.  Could you please tell me how to solve the problem?

**Comment by PragueSquare** [comment]
Thank you very much for your answer. I read your paper "Seeding strategies in search-based unit test generation" and tried to use "constant seeding" strategy. However, I ran into a problem.

As the following example,
if (x > 10) {
...
}
How do we collect the constant of "10"?

The paper says that the constants can be obtained by bytecode, but is there a value "10" in the constant pool of the bytecode? I can get the variable "x" through ASM now, but I can't get the value "10". Could you please tell me how to solve the problem?

**Comment by arcuri82** [comment]
hi,
 there would be a bytecode instruction that pushes the constant 10 on the JVM stack. However, this forum is for questions about EvoMaster (bugs and feature requests)... not questions on how to use ASM :)  (unless you were planning to make a Pull Request on EvoMaster). For that kind of questions, better to ask directly the developers of ASM (or general forums like StackOverflow)

**Comment by PragueSquare** [comment]
Yeah, you are right :) Thank you while I hava another question. Generally speaking, if we use fitness evaluation as the stopping criterion,  how many fitness evaluations are appropriate?

**Comment by arcuri82** [comment]
it is totally problem dependent... usually you would use "time" as stopping crieterion. "fitness evaluations" is only for when you running experiments. I have used 10k, 100k and 300k in the past, but again, it is problem dependent...
